enhancements/dns/externaldns-operator-aws-assume-role.md

```diff
@@ -0,0 +1,287 @@
+
+The updates to the ExternalDNS Operator are decoupled from the standard release schedule of OpenShift Container
+Platform. Consequently, in a shared VPC cluster using cross-account DNS record creation with the Ingress Operator, there
+is a possibility of installing a version of the ExternalDNS Operator that lacks support for this same feature. However,
+it is important to note that the ExternalDNS Operator and the installer or Ingress Operator do not rely on each other's 
+logic. Hence, this mismatch does not pose a concern.
+
+### Operational Aspects of API Extensions
+
+N/A
+
+#### Failure Modes
+
+- If the role ARN is not configured with the appropriate permissions, ExternalDNS will error and be unable to generate
+  DNS records.
+
+#### Support Procedures
+
+TODO
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation History`.
+
+## Alternatives
+
+### Bumping the API
+
+The appropriate fix for addressing the problem outlined in [The Required Credentials Issue](#the-required-credentials-issue)
+would be as follows:
+1. Update the CRD schema to make the existing `Credentials` field optional.
+2. Create a new version of the API that makes the optional field a pointer (assuming `Credentials` unset and `AssumeRole`
+   set is a valid combination).
+3. Update new API to not allow zero value of `Credentials`.
+4. Make the conversion webhook convert new/nil to old/zero-value and old/zero-value to new/nil, and new/zero-value to
+   old/zero-value.
+
+However, the concern with bumping the API solely for this update is that the process of bumping can be quite
+time-consuming, and it doesn't align the timeline of this enhancement. Additionally, it would be more efficient to
+bundle this fix with other API updates, such as the need to rebase for incorporating an upstream fix related to the
+Infoblox view parameter (https://github.com/kubernetes-sigs/external-dns/pull/3301) at a later point.
+
+### Assume Role Strategy
+
+Assuming a different AWS role could be implemented without ExternalDNS's built in `--aws-assume-role` as demonstrated in
+https://github.com/openshift/external-dns-operator/pull/191 by using kiam, IRSA, or kube2iam. These are alternate
+methods for assuming a different AWS role, which include utilizing third-party software rather than interacting with
+ExternalDNS directly. Therefore, this method was not chosen.
+
+### External ID
+
+We considered supporting configuration of an [external ID](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html)
+for the ExternalDNS Operator when using an assumed role. ExternalDNS provides support for an external ID via the [`--aws-assume-role-external-id`](https://github.com/kubernetes-sigs/external-dns/blob/22da9f231dbc6faa3a668b507a4a06823a129609/pkg/apis/externaldns/types.go#L460)
+command line argument. AWS suggests the use of an external ID in specific scenarios to mitigate the confused deputy
```

Comment:
Add a link for the `confused deputy` problem.  I thought you already did that?  Are you missing a commit?

Reply:
Yep, I caught it too while I was reviewing old comments. I thought I did it, but not sure what happened. Added.